lib/tesla/middleware/logger.ex

```diff
@@ -161,8 +160,13 @@ defmodule Tesla.Middleware.Logger do
 
     config = Keyword.merge(@config, opts)
 
+    format =
+      config
+      |> Keyword.get(:format)
+      |> Formatter.compile()
+
```

Comment:
I think we should do something like this:

```elixir
format = if opts[:format], do: Formatter.compile(opts[:format]), else: @format
```

so in case of not using runtime config we still benefit from compile-time format compilation.

Reply:
@teamon 

I updated the code to use compile-time format. when there is no runtime format given in the option.
but we can't use `opts[:format]` since it's keyword not a map. So will have to use `Keyword.get/2` instead